Docs: examples for io streams with StreamConverters (#28746)
akka/akka: akka-docs/src/main/paradox/stream/operators/StreamConverters/fromOutputStream.md

@ref[Additional Sink and Source converters](../index.md#additional-sink-and-source-converters)

## Signature

@apidoc[StreamConverters.fromOutputStream](StreamConverters$) { scala="#fromOutputStream(out:()=%3Cjava.io.OutputStream,autoFlush:Boolean):akka.stream.scaladsl.Sink[akka.util.ByteString,scala.concurrent.Future[akka.stream.IOResult]]" java="#fromOutputStream(akka.japi.function.Creator)" }

## Description

> Creates a Sink which writes incoming `ByteString`s to an `OutputStream` created by the given function.

Reviewer: ```suggestion
Creates a Sink which writes incoming @apidoc[ByteString]s to a `java.io.OutputStream` created by the given function.
```

> Materializes a @java[`CompletionStage`]@scala[`Future`] of `IOResult` that will be completed with the size of the file (in bytes) at the streams completion,

Reviewer: ```suggestion
Materializes a @java[`CompletionStage`]@scala[`Future`] of `IOResult` that will be completed with the size of the file (in bytes) on completion,
```

Author: I'm not 100% convinced about this change. And this writeing is used in multiple places (I've found the exact match in about 10 other locationsin the code). Maybe fix it in a separate PR?

> and a possible exception if IO operation was not completed successfully.

Reviewer: This is one of those operators that doesn't fail the stream on error, isn't it? That would be great to add a warning about note.

Author: I'm not sure. I went searching for such info in api docs and even dug in [the code](https://github.com/akka/akka/blob/6bf20f4117a8c27f8bd412228424caafe76a89eb/akka-stream/src/main/scala/akka/stream/impl/io/OutputStreamGraphStage.scala#L40-L47) and it looks like the stage is failed when the IO stream throws an exception. At least in some cases, I'm not sure the code is complete.

Reviewer: That was [fixed in 2.6](https://doc.akka.io/docs/akka/current/project/migration-guide-2.5.x-2.6.x.html#iosources-fileio)
